source/game.ai.bmx

```diff
@@ -2133,12 +2165,15 @@ endrem
 	Method rb_GetSignAtIndex:TLuaFunctionResult(arrayIndex:Int = -1)
 		If Not _PlayerInRoom("roomboard") Then Return TLuaFunctionResult.Create(Self.RESULT_WRONGROOM, Null)
 
-		Local Sign:TRoomBoardSign = GetRoomBoard().GetSignAtIndex(arrayIndex)
-		If Sign
-			Return TLuaFunctionResult.Create(Self.RESULT_OK, Sign)
-		Else
-			Return TLuaFunctionResult.Create(Self.RESULT_NOTFOUND, Null)
-		EndIf
+		Try
```

Comment:
Die Signs sind ja das Spiel ueber "konstant" vorhanden (da kommen keine neuen hinzu oder alte verschwinden). Entsprechend sollte "count()" dort immer einen festen Wert zurueckgeben und "GetSignAtIndex" keinen "Out of bounds" hervorrufen.

Ein Multithread-Problem kann m.M.n nur entstehen, wenn man ein Objekt (bzw den Pointer) aus einem Array holt, waehrend ein anderer Thread diesen Arrayeintrag ueberarbeiten will... Pointer sind halt 64 bit und nur 8 bit (1 Byte) wird garantiert atomic gespeichert/gelesen.

Ich weiss nicht, ob in dem Fall ein "not found" so die richtige Loesung ist.

Reply:
Ich hatte erstmal blind nach allen indexAt gesucht und mit demselben Pattern behandelt.

Reply:
Naja die Unsicherheit, ob das "die richtige Loesung" ist, besteht auch bei den anderen Aenderungen.

Du schuetzt damit zwar vlt. vor einem Crash, lieferst der KI aber fehlerhafte Daten. Es koennte zu "truegerischem Sicherheitsgefuehl" fuehren.

Und wenn wir die Teile am Ende eh alle mutexen muessten, waeren die Try-Catch an sich nicht mehr noetig - oder?

Reply:
Vorläufig bekämen wir (wenn das Exceptionhandling wie erwartet funktioniert) aber zumindest Log-Einträge, an welchen Stellen es zu einem Absturz gekommen wäre.
Bei den IndexAt-Aufrufen gibt es im Lua-Code ohnehin oft noch einen null-check. Die KI geht also sowieso nicht immer von richtigen Daten aus.

Ich habe kein Problem, den PR erstmal nochliegen zu lassen, bis Du Deine Härtungeng gemacht hast.